packages/twenty-server/src/engine/core-modules/external-event/services/external-event.service.ts

```diff
@@ -0,0 +1,94 @@
+import { Injectable } from '@nestjs/common';
+
+import { ClickHouseService } from 'src/database/clickHouse/clickHouse.service';
+import { TwentyConfigService } from 'src/engine/core-modules/twenty-config/twenty-config.service';
+
+import { EventMetadataService } from './event-metadata.service';
+
+/**
+ * Interface for external event input, matching the ClickHouse schema
+ */
+export interface ExternalEventInput {
+  // The event name/type (maps to 'event' column in ClickHouse)
+  event: string;
+
+  // ID of the object related to this event
+  objectId: string;
+
+  // Type of the object related to this event (e.g., 'company', 'person', 'opportunity')
+  objectType?: string;
+
+  // User ID related to this event (optional)
+  userId?: string;
```

Comment:
I think instead of "objectId"/"objectType", we want to have recordId (which is the id of the instance), and objectMetadataId which points to the metadata table